src/main/java/net/sf/jabref/logic/exporter/BibtexDatabaseWriter.java

```diff
@@ -137,6 +139,25 @@ protected void writePrelogue(BibDatabaseContext bibDatabaseContext, Charset enco
             getWriter().write("% ");
             getWriter().write(SavePreferences.ENCODING_PREFIX + encoding);
             getWriter().write(OS.NEWLINE);
+
+        } catch (IOException e) {
+            throw new SaveException(e);
+        }
+    }
+
+    @Override
+    protected void writeDatabaseID(String databaseID) throws SaveException {
+        try {
+            if (Objects.nonNull(databaseID)) {
```

Comment:
Please don't use null values for databaseID. (also check for empty string, StringUtils.isBlank)